backend/vertical-healthcare_ch/forum_datenaustausch_ch.invoice_450_request/src/main/java/de/metas/vertical/healthcare_ch/forum_datenaustausch_ch/invoice_450/Invoice450FromCrossVersionModelTool.java

```diff
@@ -1584,9 +1589,27 @@ private XmlEsr createEsrQrFromEsr9(final @NonNull XmlEsr9 esr9, @NonNull final S
 		return xEsrQR.build();
 	}
 
-	private boolean hasEsr9(final XmlRequest xAugmentedRequest)
+	/**
+	 * This is needed because the only valid candidates that can be converted to EsrQR are:
+	 * <ol>
+	 *     <li>those of type Esr9</li>
+	 *     <li>those that contain an {@code XmlBank} definition</li> (because Bank is mandatory for EsrQR)
+	 *
+	 * @param xAugmentedRequest the request to check
+	 * @return true if the request
+	 * <ul>
+	 *   <li>has an XmlEsr that is of type Esr9</li>
+	 *   <li>this Esr9 has a {@code XmlBank} defined</li>
+	 */
+	private boolean hasValidEsr9ConversionCandidate(final XmlRequest xAugmentedRequest)
 	{
-		return xAugmentedRequest.getPayload().getBody().getEsr() instanceof XmlEsr9;
+		final XmlEsr esr = xAugmentedRequest.getPayload().getBody().getEsr();
+		final boolean result = esr instanceof XmlEsr9 && ((XmlEsr9)esr).getBank() != null;
+		if (!result && esr instanceof XmlEsr9)
+		{
+			Loggables.withLogger(logger, Level.DEBUG).addLog("ESR9 -> EsrQR conversion not possible because bank record not defined in Esr9, but is mandatory in EsrQR: ");
```

Comment:
```suggestion
			Loggables.withLogger(logger, Level.DEBUG).addLog("ESR9 -> EsrQR conversion not possible because bank record not defined in Esr9, but is mandatory in EsrQR");
```